discovery/gossiper.go

```diff
@@ -888,8 +896,33 @@ func (d *AuthenticatedGossiper) findGossipSyncer(pub *btcec.PublicKey) *gossipSy
 		return syncer
 	}
 
-	// Otherwise, we'll obtain the mutex, then check again if a gossiper
-	// was added after we dropped the read mutex.
+	// A known gossip syncer doesn't exist, so we may have to create one
+	// from scratch. To do so, we'll query for a reference directly to the
+	// active peer.
+	syncPeer, err := d.cfg.FindPeer(pub)
+	if err != nil {
+		log.Debugf("unable to find gossip peer for pub=%v: %v",
```

Comment:
If we can't find the peer, then this should exit all together. As this means the peer is no longer even present in the server. Allowing the gossipSyncer to go through the server again (the second case below) doesn't fix the problem. We need it to _always_ just contact the server directly.